In [2]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.formula.api as smf
import numpy as np

In [3]:
import pandas as pd 
cars = pd.read_csv("Toyoto_Corrola.csv")
cars.head()

Id                                           Model  Price  Age_08_04  \
0   1   TOYOTA Corolla 2.0 D4D HATCHB TERRA 2/3-Doors  13500         23   
1   2   TOYOTA Corolla 2.0 D4D HATCHB TERRA 2/3-Doors  13750         23   
2   3  ÊTOYOTA Corolla 2.0 D4D HATCHB TERRA 2/3-Doors  13950         24   
3   4   TOYOTA Corolla 2.0 D4D HATCHB TERRA 2/3-Doors  14950         26   
4   5     TOYOTA Corolla 2.0 D4D HATCHB SOL 2/3-Doors  13750         30   

      KM  HP  Doors  Cylinders  Gears  Weight  
0  46986  90      3          4      5    1165  
1  72937  90      3          4      5    1165  
2  41711  90      3          4      5    1165  
3  48000  90      3          4      5    1165  
4  38500  90      3          4      5    1170

In [4]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1436 entries, 0 to 1435
Data columns (total 10 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Id         1436 non-null   int64 
 1   Model      1436 non-null   object
 2   Price      1436 non-null   int64 
 3   Age_08_04  1436 non-null   int64 
 4   KM         1436 non-null   int64 
 5   HP         1436 non-null   int64 
 6   Doors      1436 non-null   int64 
 7   Cylinders  1436 non-null   int64 
 8   Gears      1436 non-null   int64 
 9   Weight     1436 non-null   int64 
dtypes: int64(9), object(1)
memory usage: 112.3+ KB


In [5]:
cars1 = cars.drop(columns=["Model","Id"])
cars1

Price  Age_08_04     KM   HP  Doors  Cylinders  Gears  Weight
0     13500         23  46986   90      3          4      5    1165
1     13750         23  72937   90      3          4      5    1165
2     13950         24  41711   90      3          4      5    1165
3     14950         26  48000   90      3          4      5    1165
4     13750         30  38500   90      3          4      5    1170
...     ...        ...    ...  ...    ...        ...    ...     ...
1431   7500         69  20544   86      3          4      5    1025
1432  10845         72  19000   86      3          4      5    1015
1433   8500         71  17016   86      3          4      5    1015
1434   7250         70  16916   86      3          4      5    1015
1435   6950         76      1  110      5          4      5    1114

[1436 rows x 8 columns]

In [7]:
cars1.rename(columns = {'Age_08_04':'Age'}, inplace = True)
cars1

Price  Age     KM   HP  Doors  Cylinders  Gears  Weight
0     13500   23  46986   90      3          4      5    1165
1     13750   23  72937   90      3          4      5    1165
2     13950   24  41711   90      3          4      5    1165
3     14950   26  48000   90      3          4      5    1165
4     13750   30  38500   90      3          4      5    1170
...     ...  ...    ...  ...    ...        ...    ...     ...
1431   7500   69  20544   86      3          4      5    1025
1432  10845   72  19000   86      3          4      5    1015
1433   8500   71  17016   86      3          4      5    1015
1434   7250   70  16916   86      3          4      5    1015
1435   6950   76      1  110      5          4      5    1114

[1436 rows x 8 columns]

In [9]:
cars1.isna().sum()

Price        0
Age          0
KM           0
HP           0
Doors        0
Cylinders    0
Gears        0
Weight       0
dtype: int64

In [11]:
cars1[cars1.duplicated(keep=False)]

Price  Age     KM   HP  Doors  Cylinders  Gears  Weight
112  24950    8  13253  116      5          4      5    1320
113  24950    8  13253  116      5          4      5    1320

#### Obserations
- There are no NUll values
- -There are duplicated rows with indx112 ans 113
- The continuous variable columns: Price,Age,KM, HP and Weight
- The categorical columns are Doors, Cylinders and Gears
- The price column is the predicted (y) var 

In [ ]:
       v v v v v v v v v v v v v v 